src/main/java/org/aerogear/kryptowire/KryptowireServiceImpl.java

```diff
@@ -58,4 +71,59 @@ public JSONObject submit(String platform, FilePath filePath) throws IOException,
         return new JSONObject(responseBody);
     }
 
+    @Override
+    public AnalysisStatus getStatus(String hash) throws IOException, InterruptedException {
+        String endPointUrl = getApiEndpoint() + "/api/status?key=" + getApiKey() + "&hash=" + hash;
+
+        HttpGet get = new HttpGet(endPointUrl);
+
+        HttpClient httpclient = HttpClients.createDefault();
+
+        ResponseHandler<String> handler = new BasicResponseHandler();
+        String responseBody = httpclient.execute(get, handler);
+
+        JSONObject res = new JSONObject(responseBody);
+
+        String status = res.getString("status");
+
+        if (status.equals("processing")) {
+            return AnalysisStatus.PROCESSING;
+        }
+
+        if (status.equals("not_submitted")) {
+            return AnalysisStatus.NOT_SUBMITTED;
+        }
+
+        return AnalysisStatus.COMPLETE;
+    }
+
+    @Override
+    public JSONObject getResult(String uuid) throws IOException, InterruptedException {
+        String endPointUrl = getApiEndpoint() + "/api/submitted-apps?key=" + getApiKey();
```

Comment:
this is the only method that has all the fields we need such as "complete" and "submitted_at" + we can filter by the analysis unique id (uuid field)